Sources/Rendering/Core/Camera/index.js

```diff
@@ -343,9 +343,67 @@ function vtkCamera(publicAPI, model) {
   publicAPI.setObliqueAngles = (alpha, beta) => {};
   publicAPI.getOrientation = () => {};
   publicAPI.getOrientationWXYZ = () => {};
-  publicAPI.getFrustumPlanes = (aspect) => {
-    // Return array of 24 params (4 params for each of 6 plane equations)
+
+  publicAPI.getFrustumPlanes = (aspect = 1.0) => {
+    // Allocate normals as a flat array (6 planes * 4 components)
+    const normals = new Float64Array([
+      // Left
+      1, 0, 0, 1,
+      // Right
+      -1, 0, 0, 1,
+      // Bottom
+      0, 1, 0, 1,
+      // Top
+      0, -1, 0, 1,
+      // Near
+      0, 0, 1, 1,
+      // Far
+      0, 0, -1, 1,
+    ]);
+
+    const planes = new Float64Array(24);
+
+    // Get the composite projection matrix
+    const matrix = publicAPI.getCompositeProjectionMatrix(aspect, -1, 1);
+
+    // Transpose the matrix for use with normals
+    const transposed = new Float64Array(16);
+    for (let row = 0; row < 4; row++) {
+      for (let col = 0; col < 4; col++) {
+        transposed[row * 4 + col] = matrix[col * 4 + row];
+      }
+    }
+
+    // Transform the normals to world coordinates
+    const transformed = new Float64Array(4);
+    for (let i = 0; i < 6; i++) {
+      // MultiplyPoint: matrix * n
+      for (let row = 0; row < 4; row++) {
+        transformed[row] =
+          transposed[row * 4 + 0] * normals[i * 4 + 0] +
+          transposed[row * 4 + 1] * normals[i * 4 + 1] +
+          transposed[row * 4 + 2] * normals[i * 4 + 2] +
+          transposed[row * 4 + 3] * normals[i * 4 + 3];
+      }
+
+      // Normalize
+      const f =
+        1.0 /
+        Math.sqrt(
+          transformed[0] * transformed[0] +
+            transformed[1] * transformed[1] +
+            transformed[2] * transformed[2]
+        );
+
+      planes[4 * i + 0] = transformed[0] * f;
+      planes[4 * i + 1] = transformed[1] * f;
+      planes[4 * i + 2] = transformed[2] * f;
+      planes[4 * i + 3] = transformed[3] * f;
```

Comment:
you should be able to use `vtkMath.normalize()` I believe

Reply:
vtkMath.normalize doesn't have impact on the 4th component

Reply:
We could add a `vtkMath::normalize4D`...

Reply:
i will add it

Reply:
let's give it another try